src/mrnet/network/reaction_network.py

```diff
@@ -1432,11 +1439,95 @@ def generate_node_string(combined_reactants, combined_products):
                 node_str = str(combined_reactants[0]) + "," + str(combined_products[0])
         return node_str
 
+    @staticmethod
+    def concerted_reaction_filter(in_reaction_node, out_reaction_node):
+        """
+        A method to identify a valid concerted reaction based on stiochomtery of maximum of 2 reactants and products
+        :param in_reaction_node: incoming reaction node "1+PR_2,3"
+        :param out_reaction_node: outgoing reaction node "6,2+7"
```

Comment:
yes you are right, i was trying to show the cancellation concept and didnt think through the example. Switching the incoming and outgoing makes sense. Other way around would not be captured since both edges are coming into 2.